### DISCUSSION TITLE
Dependencies from galaxy.yml are not picked up.

### DISCUSSION THREAD

**Discussion by tisoft** [renovatebot/renovate#27901] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, 37.240.1

### Please tell us more about your question or problem

The galaxy.yml files are found and marked as relevant for ansible-galaxy.

But then the manager only reports 1 file. It is the requirements.yml file, dependency reolution does work for this file. Dependencies from the */galaxy.yml files are not detected.

It's also not working in hosted renovate on github: https://github.com/advanova/renovate_galaxy_reproducer

### Logs (if relevant)

<details><summary>Logs</summary>

```

{"name":"renovate","hostname":"runner-orkj8jvs-project-13-concurrent-0","pid":53,"level":20,"logContext":"1H11fIyY_ewlfZx8Xw4pc","repository":"advanova/ansible/advanova-ansible-collections","msg":"Matched 5 file(s) for manager ansible-galaxy: base/galaxy.yml, devops/galaxy.yml, requirements.yml, vendor/galaxy.yml, vmobil/galaxy.yml","time":"2024-03-12T07:03:56.120Z","v":0}

...

{"name":"renovate","hostname":"runner-orkj8jvs-project-13-concurrent-0","pid":53,"level":30,"logContext":"1H11fIyY_ewlfZx8Xw4pc","repository":"advanova/ansible/advanova-ansible-collections","baseBranch":"main","stats":{"managers":{"ansible-galaxy":{"fileCount":1,"depCount":1},"asdf":{"fileCount":1,"depCount":2},"dockerfile":{"fileCount":1,"depCount":2},"github-actions":{"fileCount":1,"depCount":3},"gitlabci":{"fileCount":1,"depCount":1},"gitlabci-include":{"fileCount":1,"depCount":1},"maven":{"fileCount":1,"depCount":4},"pip_requirements":{"fileCount":1,"depCount":9},"pipenv":{"fileCount":1,"depCount":11}},"total":{"fileCount":9,"depCount":34}},"msg":"Dependency extraction complete","time":"2024-03-12T07:03:56.157Z","v":0}
```
</details>

**Comment by rarkins** [comment]
Is it specific to the "test" folder? What if you rename it?

**Reply by tisoft** [reply]
I have renamed it in the reproducer.

The hosted renovate does find the files, but doesn't detect any dependencies in it.

```
DEBUG: Matched 2 file(s) for manager ansible-galaxy: advanova/vmobil/galaxy.yml, galaxy.yml
DEBUG: manager extract durations (ms)
{
  "managers": {
    "ansible-galaxy": 11
  }
}
```

**Reply by rarkins** [reply]
And what if you have the exact same galaxy.yml contents in both - does it find dependencies in both?

**Reply by tisoft** [reply]
It does not find any dependencies in any of the two galaxy.yml files.

I have now added a requirements.yml file, the same dependencies are recognized there.

**Reply by rarkins** [reply]
Could you clarify what you mean?

**Reply by tisoft** [reply]
I can try. :)

The requirements.yml file works fine. The dependencies are detected and PRs are opened.

The same dependencies, that work in that file, are not recognized in the galaxy.yml files. Both files are detected and assigned to the ansible-galaxy manager, but the manager does not find any dependencies in them.

It should detect the same dependencys as in the requirements.yml file

**Comment by secustor** [comment, marked as answer]
Change the inline object declaration to a standard one, then it should pick it up.

**Reply by tisoft** [reply]
Thanks! That works.

Is it documented anywhere, that inline objects are not supported?

**Reply by secustor** [reply]
Not yet, PRs are welcome.
